tests/test_run_ccoverage.py

```diff
@@ -9,9 +9,13 @@
 import logging
 import unittest
 
+from _pytest.monkeypatch import MonkeyPatch
```

Comment:
Instead of importing this from `_pytest`, instantiating it, and getting a context manager, you can just use it as a [fixture](https://docs.pytest.org/en/latest/reference.html#fixtures) (no import required).

    def test_main(monkeypatch):
        monkeypatch.setattr(...)

Reply:
Apparently, in order for me to use this, my method [has to be a fixture](https://stackoverflow.com/a/37956445/445241). In this case, it isn't, so I guess I use the context manager way then.

Reply:
Did it not work? I'm using it in FuzzManager in tests (eg. [here](https://github.com/MozillaSecurity/FuzzManager/blob/master/Collector/test_Collector.py#L62)).

That SO is about using it in a setup method, which is not a test, so it has to be a fixture to use other fixtures. That doesn't apply to you.

Reply:
It works if the test_* function is not in a class:

```
def test_main(monkeypatch):
```

However, it doesn't if it is in a custom class:
```
class RunCcoverageTests(unittest.TestCase):
    """"TestCase class for functions in run_ccoverage.py"""

    @staticmethod
    /snip
    def test_main(monkeypatch):
```

It fails with the following error:

`TypeError: test_main() missing 1 required positional argument: 'monkeypatch'`

Question: Is `RunCcoverageTests` supposed to inherit from `unittest.TestCase` for things to work?

Reply:
I tried:

```
import _pytest
/snip

class RunCcoverageTests(_pytest.monkeypatch.MonkeyPatch):
    """"TestCase class for functions in run_ccoverage.py"""

    @staticmethod
    /snip
    def test_main(monkeypatch):
```

and this seems to work. Is this the right intended fix? [SO also mentions](https://stackoverflow.com/a/42334155/445241) that `unittest.TestCase` cannot be monkeypatched in the form I mentioned in the previous comment, so I think this is related.

I know this can be solved if I remove the class entirely, but I (may) have some other tests that use their own class in more elaborate ways.

Reply:
Another question: by *not* using a `with` statement, wouldn't the function continue to be monkeypatch'ed for the remaining of the runs of tests by pytest?

Reply:
Oh no, calling `_pytest.monkeypatch.MonkeyPatch` results in pytest not being able to discover this test automatically.

Reply:
Hmm, you're right. The [pytest unittest section](https://docs.pytest.org/en/latest/unittest.html) explains why in the notes at the end. Any new tests I write are not inheriting from `unittest`. The only point of it is to run existing test suites while transitioning to `pytest`.

About the monkeypatch lifetime without using a context (`with`):  Fixtures are cleaned up automatically. They are the `pytest` equivalent of `unittest.TestCase.setUp()` and `unittest.TestCase.tearDown()`. Another useful one is `tmp_path` which gives you a `pathlib.Path` object to a unique temporary directory.

So here's the `unittest` way to use a fixture like `monkeypatch` without using the internal classes in `_pytest`:
```
import time
import unittest
import pytest


@pytest.fixture
def cls_monkeypatch(request, monkeypatch):
    request.cls.monkeypatch = monkeypatch


@pytest.mark.usefixtures('cls_monkeypatch')
class ThingTest(unittest.TestCase):

    def test_something(self):
        self.monkeypatch.setattr(time, 'sleep', lambda x: None)
        start = time.time()
        time.sleep(1)
        end = time.time()
        self.assertLess(end - start, 1)
```

... and here's the `pytest` way:
```
import time


def test_something(monkeypatch):
        monkeypatch.setattr(time, 'sleep', lambda x: None)
        start = time.time()
        time.sleep(1)
        end = time.time()
        assert (end - start) < 1
```

Reply:
I'll move entirely to pytest in another PR. Thanks!